### ISSUE TITLE
Error: undefined method 'read_timeout=' for IO::ARGF (compile-time type is IO+)

### ISSUE THREAD

**Issue by robin-randhawa-sifive** [crystallabs/crysterm#14]
Hi.

I seem to be hitting an undefined method. Here are some relevant details. Any insights appreciated.

Thanks!

```
~/Work/experiments/crystal/repos
❯ git clone git@github.com:crystallabs/crysterm.git
Cloning into 'crysterm'...
remote: Enumerating objects: 3038, done.
remote: Counting objects: 100% (837/837), done.
remote: Compressing objects: 100% (218/218), done.
remote: Total 3038 (delta 616), reused 835 (delta 616), pack-reused 2201
Receiving objects: 100% (3038/3038), 1.03 MiB | 2.25 MiB/s, done.
Resolving deltas: 100% (2283/2283), done.

~/Work/experiments/crystal/repos
❯ cd crysterm/

~/Work/experiments/crystal/repos/crysterm master
❯ shards install
Resolving dependencies
Fetching https://github.com/crystallabs/crystallabs-helpers.cr.git
Fetching https://github.com/samuallb/w3mimagedisplay.git
Fetching https://github.com/crystallabs/event_handler.git
Fetching https://github.com/crystallabs/gpm.cr.git
Fetching https://github.com/crystal-ameba/ameba.git
Fetching https://github.com/crystallabs/tput.git
Fetching https://github.com/crystallabs/term_colors.git
Fetching https://github.com/crystallabs/unibilium.cr.git
Fetching https://github.com/crystallabs/unibilium-shim.cr.git
Fetching https://github.com/crystal-term/screen.git
Fetching https://github.com/crystal-posix/ioctl.cr.git
Installing unibilium (1.0.1)
Installing unibilium-shim (1.1.0)
Installing crystallabs-helpers (1.0.1)
Installing ioctl (0.1.2)
Installing term-screen (0.1.0 at ea51ee8)
Installing tput (1.0.5 at e67c113)
Installing term_colors (1.0.1)
Installing w3m_image_display (0.1.2)
Installing event_handler (1.0.3)
Installing gpm (1.0.5)
Installing ameba (0.13.4)
Postinstall of ameba: make bin && make run_file
Writing shard.lock

~/Work/experiments/crystal/repos/crysterm master
❯ crystal run examples/tech-demo.cr
Showing last frame. Use --error-trace for full trace.

In lib/tput/src/tput/input.cr:61:15

 61 | input.read_timeout = @read_timeout
            ^-----------
Error: undefined method 'read_timeout=' for IO::ARGF (compile-time type is IO+)

~/Work/experiments/crystal/repos/crysterm master
❯ crystal --version
Crystal 1.6.2 (2022-11-03)

LLVM: 14.0.6
Default target: aarch64-unknown-linux-gnu

~/Work/experiments/crystal/repos/crysterm master
❯ uname -a
Linux chiclet 6.1.0-asahi-2-2-edge-ARCH #2 SMP PREEMPT_DYNAMIC Fri, 16 Dec 2022 05:04:23 +0000 aarch64 GNU/Linux
```

**Comment by docelic** [comment]
Hey,
For a quick fix you could just open `lib/tput/src/tput/input.cr` and comment/deactivate two lines where `read_timeout=` is mentioned (lines 61 and 75).
In parallel, I will commit a proper fix.

Thanks for the report, cheers!

**Comment by robin-randhawa-sifive** [comment]
Thanks @docelic, that worked just fine! I'll standby for a proper fix and chime in again if needed.

Glad you noticed the asahilinux installation! :) I _love_ it! Would highly recommend using it!

Thanks a lot once again! Have a good one!

**Comment by bararchy** [comment]
I was just hitting this exact issue as well, I hope it can be fixed soon, as this lib seems to be interesting for playing out with

**Comment by docelic** [comment]
Hey Bar, thanks, I'll do another attempt at fixing it (in the `tput` dependency, where the source of the error is).
Will update here as soon as there are news.

**Comment by bararchy** [comment]
@docelic thank you! both for the quick response and the willingness to work on this amazing lib

**Comment by docelic** [comment]
Heya, this should be fixed in tput 1.0.7. `shards update` should work!
